### ISSUE TITLE
aoco table does not support tsquery type

### ISSUE THREAD

**Issue by HelloYJohn** [greenplum-db/gpdb#15556]
## Bug Report
```
testdb=# \d test_ts3
              Table "public.test_ts3"
 Column |  Type   | Collation | Nullable | Default 
--------+---------+-----------+----------+---------
 c2     | tsquery |           |          | 
 c3     | integer |           |          | 
Checksum: t
Distributed by: (c3)

testdb=# truncate test_ts3;
TRUNCATE TABLE
testdb=# insert into test_ts3(c2 ,c3) select 'fat & rat'::tsquery, 1 ;
INSERT 0 1
testdb=# select * from test_ts3;
ERROR:  stack depth limit exceeded
HINT:  Increase the configuration parameter "max_stack_depth" (currently 2048kB), after ensuring the platform's stack depth limit is adequate.
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
The connection to the server was lost. Attempting reset: Failed.

```

### Greenplum version or build

```
testdb=# select version();
                                                                                   version                                                                                   
 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 PostgreSQL 12beta2 (Greenplum Database 7.0.0-beta.0 build devOpen Source) on aarch64-unknown-linux-gnu, compiled by gcc (GCC) 7.3.0, 64-bit compiled on Feb  6 2023 10:46:39
(1 row)
```

### Expected behavior
show right result
### Actual behavior
overflow

### Step to reproduce the behavior
create table test_ts3(c2 tsquery, c3 int)WITH (appendonly=true, compresstype=zlib, compresslevel=5,orientation=column) distributed by (c3);
insert into test_ts3(c2 ,c3) select 'fat & rat'::tsquery, 1 ;
select * from test_ts3;

**Comment by adam8157** [comment]
Inserting crashes on 6X.

```
postgres=# create table test(c1 tsvector, c2 tsquery, c3 int)WITH (appendonly=true, compresstype=zlib, compresslevel=5,orientation=column) distributed by (c3);
CREATE TABLE
postgres=# INSERT INTO test(c1, c2, c3) VALUES('a fat cat sat on a mat and ate a fat rat'::TSVECTOR, 'fat & rat'::TSQUERY, 1);
ERROR:  Error on receive from seg1 127.0.1.1:40001 pid=60296: server closed the connection unexpectedly
        This probably means the server terminated abnormally
        before or while processing the request.
```

**Comment by adam8157** [comment]
Fixed by #15575
